### ISSUE TITLE
[CLOSED] privateMessage permission only affects /msg

### ISSUE THREAD

**Issue by JMakey** [BZFlag-Dev/bzflag-import-7#2]
<a href="https://github.com/blast007"><img src="https://avatars.githubusercontent.com/u/328999?v=3" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [blast007](https://github.com/blast007)**
_Tuesday Jun 10, 2014 at 22:36 GMT_
_Originally opened as https://github.com/BZFlag-Dev/bzflag-import-3/issues/2_

----

The privateMessage permission only affects the use of /msg, not directly sending a targeted MsgMessage message.  So, using the arrow keys while composing a message to select a recipient or using the period or comma keys still allows you to privately message players without the privateMessage permission.

Lacking the privateMessage permission also completely disables the /msg command when some targets like >all, >admin, or >team should still be allowed.

**Comment by JMakey** [comment]
<a href="https://github.com/kongr45gpen"><img src="https://avatars.githubusercontent.com/u/720678?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [kongr45gpen](https://github.com/kongr45gpen)**
_Wednesday Jun 11, 2014 at 04:36 GMT_

----

Do we want to require both the `talk` and `privateMessage` permissions to let a user send a private message, or just `privateMessage`?

**Comment by JMakey** [comment]
<a href="https://github.com/JeffM2501"><img src="https://avatars.githubusercontent.com/u/322174?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [JeffM2501](https://github.com/JeffM2501)**
_Wednesday Jun 11, 2014 at 06:41 GMT_

----

Yes, Private Messages are a subset of talking. Those that can't talk should not be able to send any text information to any other user, other then possibly /report.
I could seen an exception to being able to reply to a private message that an admin send to the user with out talk so that admins can silence someone then talk to them about how to resolve the issue. 

Basically we want to be able to have three groups of users who can initiate the following chat messages
0) Users who can not send chat to anyone
1) Users who can send chat to anyone, public or private
2) Users who have been restricted to public only chat.

I can not see any reason to have a class of user who is restricted just to private chat.

**Comment by JMakey** [comment]
<a href="https://github.com/kongr45gpen"><img src="https://avatars.githubusercontent.com/u/720678?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [kongr45gpen](https://github.com/kongr45gpen)**
_Thursday Jun 12, 2014 at 11:25 GMT_

----

@JeffM2501 The exception you mentioned is already handled by `adminMessageSend`, which lets users talk to the admin channel or send PMs to admins, even if they don't have `talk`.
